.github/workflows/ci-qodana.yml

```diff
@@ -0,0 +1,151 @@
+
+# todo - this is a test
+
+name: Qodana
+
+on:
+  pull_request:  # pull_request_target will not work unless workflow exists in target already, plus it's dangerous
+    # todo
+    # see here: https://github.com/devicons/devicon/pull/481/files#diff-e5dbed567533ccaa21c4162c057618fea0c2de0122994a827e7d2b21a942366c
+    # basically, upload pr_number/logs to artifact
+    # download artifacts from secondary workflow and process accordingly
+    branches: [master, nightly]
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [master, nightly]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  qodana:
+    name: Qodana
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write  # required to add PR comment
+
+    steps:
+      - name: Prepare
+        id: prepare
+        if: always()
+        run: |
+          # get branch... may be blank
+          branch=${GITHUB_HEAD_REF}
+          repo=${{ github.event.repository.name }}
+          echo "repo=$repo" >> $GITHUB_OUTPUT
+
+          # check the branch variable
+          if [ -z "$branch" ]
+          then
+            echo "This is a PUSH event"
+            # use the branch name
+            destination=${{ github.event.ref_name }}
+          else
+            echo "This is a PR event"
+            # use the PR number
+            destination=${{ github.event.pull_request.number }}
+          fi
+
+          echo "destination=$destination" >> $GITHUB_OUTPUT
+
+          # prepare urls
+          base=https://${{ github.repository_owner }}.github.io
+          report_url=${base}/qodana-reports/${repo}/${destination}/
+          workflow_url=https://github.com/${{ github.repository_owner }}/${repo}/actions/runs/${{ github.run_id }}
+          echo "report_url=$report_url" >> $GITHUB_OUTPUT
+          echo "workflow_url=$workflow_url" >> $GITHUB_OUTPUT
+
+      - name: Setup PR comment
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: mshick/add-pr-comment@v2
+        with:
+          repo-token: ${{ secrets.GH_BOT_TOKEN }}
+          message: |
+            :warning: **Qodana is checking this PR** :warning:
+            Live results available [here](${{ steps.prepare.outputs.workflow_url }})
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Checkout Qodana/gh-pages repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository_owner }}/qodana-reports
+          ref: gh-pages
+          path: gh-pages
+          persist-credentials: false  # otherwise, the token used is the GITHUB_TOKEN, instead of the personal token
+          fetch-depth: 0  # otherwise, will fail to push refs to dest repo
+
+      - name: Qodana
+        id: qodana
+        uses: JetBrains/qodana-action@v2022.3.0
+        with:
+          additional-cache-hash: qodana-${{ github.sha }}
```

Comment:
`additional-cache-hash` is appended to a pre-defined string (https://github.com/JetBrains/qodana-action/blob/3fdeefa830a7634a5c04f1cd70fcfb69956137a2/src/commands/scan.yml#L41)

so, just change this to `python`